### PR TITLE
Update interproscan

### PR DESCRIPTION
**Pull request by mahesh-panchal**
<!--
# nf-core/modules pull request

Many thanks for contributing to nf-core/modules!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the master branch.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

- Bioconda installation explicitly states the database is not included and needs to be. This is missing as an input.
  ```
  ######################################                                                                                                                                                                                                                                                               
  # First time usage please README !!! #                                                                                                                                                                                                                                                               
  ######################################                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                     
  The databases are huge and consequently not shipped within this installation.                                                                                                                                                                                                                        
  Please download and install the Databases manually by following the commands below:                                                                                                                                                                                                                  
  !!! /!\ Edit the 2 first lines to match the wished version of the DB /!\ !!!                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                     
  Commands:                                                                                                                                                                                                                                                                                            
  =========                                                                                                                                                                                                                                                                                            
  # See here for latest db available: https://github.com/ebi-pf-team/interproscan or http://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/ 
  # Set versions                                                                                                                                                                                                                                                                                       
  version_major=5.59                                                                                                                                                                                                                                                                                   
  version_minor=91.0                                                                                                                                                                                                                                                                                   
  CONDA_PREFIX=/the/path/to/your/interproscan/conda/env/                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                     
  # get the md5 of the databases                                                                                                                                                                                                                                                                       
  wget http://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/${version_major}-${version_minor}/interproscan-${version_major}-${version_minor}-64-bit.tar.gz.md5                                                                                                                                             
  # get the databases (with core because much faster to download)                                                                                                                                                                                                                                      
  wget http://ftp.ebi.ac.uk/pub/software/unix/iprscan/5/${version_major}-${version_minor}/interproscan-${version_major}-${version_minor}-64-bit.tar.gz                                                                                                                                                 
  # checksum                                                                                                                                                                                                                                                                                           
  md5sum -c interproscan-${version_major}-${version_minor}-64-bit.tar.gz.md5
  # untar gz
  tar xvzf interproscan-${version_major}-${version_minor}-64-bit.tar.gz
  # remove the sample DB bundled by default
  rm -rf $CONDA_PREFIX/share/InterProScan/data/
  # copy the new db
  cp -r interproscan-${version_major}-${version_minor}/data $CONDA_PREFIX/share/InterProScan/
  
  
  INFO:
  ====
  Phobius (licensed software), SignalP, SMART (licensed components) and TMHMM use
  licensed code and data provided by third parties. If you wish to run these
  analyses it will be necessary for you to obtain a licence from the vendor and
  configure your local InterProScan installation to use them.
  (see more information in $CONDA_PREFIX/share/InterProScan/data/<db>)
  ```
- Parameters changed to long format.
- `out_ext` removed as this is optional, and code currently restricts to one output.
- version string code shortened
- Removes optional flags
- simplify bash test
- use sampleDB for tests
- Otherwise use staged data directory for interproscan   

Closes #XXX <!-- If this PR fixes an issue, please link it here! -->

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/interproscan/main.nf [hunk 1]
```diff
@@ -9,7 +9,7 @@ process INTERPROSCAN {
 
     input:
     tuple val(meta), path(fasta)
-    val(out_ext)
+    path(interproscan_database, stageAs: 'data')
```
Comment (FriederikeHanssen) [inline modules/nf-core/interproscan/main.nf:12]: meta for the database just in case?

Comment (mahesh-panchal) [reply modules/nf-core/interproscan/main.nf:12]: It's always the same database for Interproscan. It's tied to the version and build of the software. Which databases inside that folder are used are toggled by the `-appl` flag.

**Comment by mahesh-panchal** [comment]
I give up trying to bypass the hard-coded memory requirement. It's just lead to a hole of java exceptions.

For info this was what was in the `else` by the time I gave up:
```bash
    else
        echo "Running Test mode!"
        # Reduce memory for tests
        mkdir bin
        INTERPROSCAN_BIN=\$( dirname \$( which interproscan.sh ) )
        INTERPROSCAN_DIR="\$( dirname "\$INTERPROSCAN_BIN" )"
        INTERPROSCAN_PROPERTIES="\$( find "\$INTERPROSCAN_DIR" -name "interproscan.properties" )"
        INTERPROSCAN_JAR="\$( find "\$INTERPROSCAN_DIR" -name "interproscan-5.jar" )"
        cp "\$( which interproscan.sh )" bin/
        cp "\$INTERPROSCAN_PROPERTIES" .
        sed -i "/^bin\\.directory=/ s|.*|bin.directory=\$INTERPROSCAN_BIN|" interproscan.properties
        sed -i "/^data\\.directory=/ s|.*|data.directory=\$( dirname \$INTERPROSCAN_DIR )/share/data|" interproscan.properties
        sed -i '/-Xms2028M -Xmx9216M/ s|-Xms2028M -Xmx9216M|-Xms1G -Xmx2G|' bin/interproscan.sh
        sed -i "/interproscan-5.jar/ s|interproscan-5.jar|\$INTERPROSCAN_JAR|" bin/interproscan.sh
        export INTERPROSCAN_CONF=interproscan.properties
        export PATH="\$PWD/bin:\$PATH"
```

**Comment by mahesh-panchal** [comment]
Should I limit this to a stub test?

**Comment by SPPearce** [comment]
> I give up trying to bypass the hard-coded memory requirement. It's just lead to a hole of java exceptions.
> 
> For info this was what was in the `else` by the time I gave up:
> 
> ```shell
>     else
>         echo "Running Test mode!"
>         # Reduce memory for tests
>         mkdir bin
>         INTERPROSCAN_BIN=\$( dirname \$( which interproscan.sh ) )
>         INTERPROSCAN_DIR="\$( dirname "\$INTERPROSCAN_BIN" )"
>         INTERPROSCAN_PROPERTIES="\$( find "\$INTERPROSCAN_DIR" -name "interproscan.properties" )"
>         INTERPROSCAN_JAR="\$( find "\$INTERPROSCAN_DIR" -name "interproscan-5.jar" )"
>         cp "\$( which interproscan.sh )" bin/
>         cp "\$INTERPROSCAN_PROPERTIES" .
>         sed -i "/^bin\\.directory=/ s|.*|bin.directory=\$INTERPROSCAN_BIN|" interproscan.properties
>         sed -i "/^data\\.directory=/ s|.*|data.directory=\$( dirname \$INTERPROSCAN_DIR )/share/data|" interproscan.properties
>         sed -i '/-Xms2028M -Xmx9216M/ s|-Xms2028M -Xmx9216M|-Xms1G -Xmx2G|' bin/interproscan.sh
>         sed -i "/interproscan-5.jar/ s|interproscan-5.jar|\$INTERPROSCAN_JAR|" bin/interproscan.sh
>         export INTERPROSCAN_CONF=interproscan.properties
>         export PATH="\$PWD/bin:\$PATH"
> ```
> 
> This is the hard-coded line I was trying to bypass:
> 
> ```
>     /usr/local/bin/interproscan.sh: line 61:   317 Killed                  "$JAVA" -XX:ParallelGCThreads=8 -Xms2028M -Xmx9216M $PROPERTY -jar interproscan-5.jar $@ -u $USER_DIR
> ```
> 
> `PROPERTY` is set within the script to the path of the `interproscan.properties`.

Where is the scream emoji in github?
So the problem is that it is hardcoded to require too much memory?

**Comment by mahesh-panchal** [comment]
For the self-hosted docker runner yes.

**Comment by SPPearce** [comment]
> For the self-hosted docker runner yes.

Then I think it is reasonable to exclude it from the docker, and just test singularity and conda.
Alternatively a stub-only test is fine.

**Comment by SPPearce** [comment]
Could potentially add a condition to catch if the task memory is less than the hardcoded expectation?

**Comment by mahesh-panchal** [comment]
Won't that break it always on test mode where the upper limit is set to 6GB?

**Comment by mahesh-panchal** [comment]
> > For the self-hosted docker runner yes.
> 
> Then I think it is reasonable to exclude it from the docker, and just test singularity and conda. Alternatively a stub-only test is fine.

We can exclude running docker? I thought that was only conda.

**Comment by SPPearce** [comment]
> > > For the self-hosted docker runner yes.
> > 
> > 
> > Then I think it is reasonable to exclude it from the docker, and just test singularity and conda. Alternatively a stub-only test is fine.
> 
> We can exclude running docker? I thought that was only conda.

You can exclude any of the three in https://github.com/nf-core/modules/blob/master/.github/workflows/test.yml#L512, just set the profile appropriately.

**Comment by SPPearce** [comment]
> Won't that break it always on test mode where the upper limit is set to 6GB?

Well yes, presumably. Could maybe add a warning rather than a fail.

**Comment by mahesh-panchal** [comment]
It's only the test profile that brings it under memory requirements so I'm not sure it's worth adding code for it. If it's a warning it's not likely to be seen as it'll get lost in the log output, and most people won't check unless it's an error.

File: modules/nf-core/interproscan/tests/main.nf.test [hunk 2]
```diff
@@ -8,58 +8,90 @@ nextflow_process {
     tag "modules_nfcore"
     tag "interproscan"
 
-    test("Annotates set of input proteins in an output tsv file") {
+    // test("sarscov2 - proteome_fasta") {
```
Comment (SPPearce) [inline modules/nf-core/interproscan/tests/main.nf.test:11]: Can you add a comment saying why they have been commented out.

**Review by SPPearce — APPROVED** [review]

File: modules/nf-core/interproscan/tests/main.nf.test (hunk 2 above)
Comment (mahesh-panchal) [reply modules/nf-core/interproscan/tests/main.nf.test:11]: done

**Comment by mahesh-panchal** [comment]
Why do some jobs only start failing when someone has hit approve?

**Pull request merged**